plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -38,6 +38,8 @@ type Query struct {
 	Script         string
 	ResultByRow    bool
 	OrderedColumns []string
+	HasCachedData  bool
+	DataCache      *QueryDataCache
```

Comment:
Btw. do we really need those exported?

Reply:
What do you mean under **exported**? If you need more details about the purpose of DataCache you can get them from the following comment [https://github.com/influxdata/telegraf/pull/9150#discussion_r652240508](https://github.com/influxdata/telegraf/pull/9150#discussion_r652240508)

Reply:
I mean that those struct members are visible and accessible outside of the plugin if starting with an uppercase letter (see https://golang.org/doc/effective_go#names). To prevent (accidental) messing with those members, you should make them start with a lowercase-letter.